### PR TITLE
Refactored fxa-client

### PR DESCRIPTION
**Pull request by bendk**
This was part of my work for the auth PR, but I think it's useful separately.

- Updated `fxa-client` to be a single binary.
- Added a cargo alias so it can be run with `cargo fxa [args]`.

I think this makes it easier to use and also will make it easier to expand the functionality.  It's easier to add another clap subcommand than to add a new binary.

### Pull Request checklist ###

- **Breaking changes**:  This PR follows our [breaking change policy](https://github.com/mozilla/application-services/blob/main/docs/howtos/breaking-changes.md)
  - [ ] This PR follows the breaking change policy:
     - This PR has no breaking API changes, or
     - There are corresponding PRs for our consumer applications that resolve the breaking changes and have been approved
- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGELOG.md](../CHANGELOG.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

[Branch builds](https://github.com/mozilla/application-services/blob/main/docs/howtos/branch-builds.md): add `[firefox-android: branch-name]` to the PR title.

### CODE REVIEW

**Comment by bendk** [comment]
AFAICT, dependency check failure is related to these changes.  When I look at `Cargo.lock`, this PR just removes a couple dependencies.

File: components/fxa-client/src/lib.rs [hunk 1]
```diff
@@ -60,9 +60,9 @@ pub use push::{
 pub use token::{AccessTokenInfo, AuthorizationParameters, ScopedKey};
 
 /// Result returned by internal functions
-type Result<T> = std::result::Result<T, Error>;
+pub type Result<T> = std::result::Result<T, Error>;
```
Comment (tarikeshaq) [inline components/fxa-client/src/lib.rs:63]: Does the internal error need to be `pub`? Looks like all calls use the public API right?

**Review by tarikeshaq — APPROVED** [review]
Awesome! This looks great - +1 to using one binary for all FxA related examples 🚢

**Comment by tarikeshaq** [comment]
Dependency check should be fixed in https://github.com/mozilla/application-services/pull/5680

**Pull request merged**